### PR TITLE
Event Statistics: track counts of YK events by type and change-type

### PR DESCRIPTION
**Pull request by richscott**
Keep a set of basic statistical counts of events emitted from the yunikorn-core event stream in a map, with the key being the event's Type + ChangeType, and the value being the number of occurrences of events with that key combo.

Also introduce a new `/ws/v1/event-statistics` endpoint, which returns the current event counts in JSON-format.

This is intended to be used with submissions of different permutations of YK jobs, so we can determine minimum/ average/ maximum numbers of events that are emitted by YK for a given application volume. In the future, we may expand this data into a larger set of statistics, depending on other data.

Also fix many linting errors.

Fixes https://github.com/G-Research/yunikorn-history-server/issues/21

### CODE REVIEW

**Review by pavlovic-ivan — APPROVED** [review]

File: internal/repository/postgres.go [hunk 1]
```diff
@@ -133,7 +133,8 @@ func (s *RepoPostgres) Setup(ctx context.Context) {
 			nodes_util_list JSONB,
 			UNIQUE (id),
 			PRIMARY KEY (id))`,
-		// for yunikorn-core/pkg/webservice/dao/ContainerHistoryDAOInfo and yunikorn-core/pkg/webservice/dao/ApplicationHistoryDAOInfo
+		// for yunikorn-core/pkg/webservice/dao/ContainerHistoryDAOInfo and
```
Comment (dave-gantenbein) [inline internal/repository/postgres.go:136]: do we need these comments?

File: internal/ykclient/api_handlers.go [hunk 2]
```diff
@@ -139,18 +163,20 @@ func (c *Client) GetApplications(ctx context.Context, partitionName, queueName s
...
 		}
 	}
 
 	return apps, nil
 }
 
-// GetApplication calls the YuniKorn Scheduler API to get the application information for the given partition, queue and appID
-// If partitionName is empty, it defaults to "default"
+// GetApplication calls the YuniKorn Scheduler API to get the application information for the given
+// partition, queue and appID. If partitionName is empty, it defaults to "default".
```
Comment (dave-gantenbein) [inline internal/ykclient/api_handlers.go:175]: Could we express this comment in the form of a unit test?

File: internal/ykclient/client.go [hunk 3]
```diff
@@ -36,50 +39,88 @@ func NewClient(httpProto string, ykHost string, ykPort int, repo *repository.Rep
...
+				fmt.Fprintf(os.Stderr, "Error: could not read from http stream: %v", err)
+				break
+			}
+
+			ev := si.EventRecord{}
+			err = json.Unmarshal(line, &ev)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "could not unmarshal event from stream: %v\n", err)
+				break
+			}
```
Comment (dave-gantenbein) [inline internal/ykclient/client.go:100]: Can we capture this in a ticket?

**Review by dave-gantenbein — CHANGES_REQUESTED** [review]

File: internal/repository/postgres.go (hunk 1 above)
Comment (richscott) [reply internal/repository/postgres.go:136]: Not really needed, the yunikorn-core structs are referenced in the postgres repo layer, in the SQL upsert logic.
Removed in commit 7927a9d

File: internal/ykclient/api_handlers.go (hunk 2 above)
Comment (richscott) [reply internal/ykclient/api_handlers.go:175]: I will create a test.

Comment (richscott) [reply internal/ykclient/api_handlers.go:175]: Test added in commit c3a3e73 - see `internal/ykclient/api_handlers_test.go`.

File: internal/ykclient/client.go (hunk 3 above)
Comment (richscott) [reply internal/ykclient/client.go:100]: This is now described in ticket https://github.com/G-Research/yunikorn-history-server/issues/34

**Review by pavlovic-ivan — APPROVED** [review]
Looking good (from my side). I reviewed only Ci related files

**Review by dave-gantenbein — APPROVED** [review]

**Pull request merged**